### ISSUE TITLE
Job name to adapt variable

### ISSUE THREAD

**Issue by Iyappanj** [ansible/awx#14733]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx/issues) for duplicates.
- [X] I understand that AWX is open source software provided for free and that I might not receive a timely response.

### Feature type

New Feature

### Feature Summary

We do see that the job names are represented now as  "Job Number- Job Template Name"
We would like to know if there is any possibility of appending some variables to it, because at time we see it difficult to search for a job that has run with some details.

Basically we have same job template running several times with different data, So at a point to search for the job that run with some specific details we had to go through every job and identify. Instead if had the option of appending variable to the job name it will help us giving some unique name for each job. So it would be easy for us to filter it in the future.

### Select the relevant components

- [X] UI
- [ ] API
- [ ] Docs
- [ ] Collection
- [ ] CLI
- [ ] Other

### Steps to reproduce

this is a feature request

### Current results

"Job Number- Job Template Name"

### Sugested feature result

"Job Number- Job Template Name- Some Variable"

**Comment by djyasin** [comment]
@Iyappanj We do have the option to prompt on launch for a job label. In the UI when you create or edit a job template you should see a field titled "labels." Beside it, you should see a check box that says "Prompt on Launch." If you check that box, when your job launches you can enter the desired label name. After your job has run, you can then query the API for your job labels. For example, we added a label called: "testing_labels" and then used this structure to query the API for jobs with that given label: `/api/v2/jobs/?labels__name=testing_labels`

Does this resolve this issue for you?

**Comment by bobumba** [comment]
@djyasin That seems to me to be the correct solution.  Labels can also be seen in the "expanded" view of jobs in the UI and searched on.  However, as the report points out, the job id and the job name are concatenated together under the name column in the UI.  This incorrectly represents the job's name field in the UI.  The job overview page in the UI should have a jobid column to correctly represent api/v2/jobs.

**Comment by Iyappanj** [comment]
@djyasin @bobumba thanks for taking your time and providing me the solution.  I will give it a try.

**Comment by Iyappanj** [comment]
Hi @djyasin @bobumba I tested it and it worked as expected. Thanks for the support.

I have a workflow template configured and I need to be able to pass the label for it dynamically through API call. So I decided to use workflow launch so that I can pass label as variable to it.  But from doc - https://docs.ansible.com/ansible/devel/collections/awx/awx/workflow_launch_module.html#ansible-collections-awx-awx-workflow-launch-module I see it is not possible to pass label as a variable here. Any suggestion from you would be much appreciated.

**Comment by Bobby-Math** [comment]
Hi @Iyappanj 

Workflow templates do not support dynamically assigning unique names to jobs as soon as they are launched in the nodes that you are managing, according to the best of my knowledge.

Job templates do support that feature, given that you want to assign a unique name of your choice to the jobs running in those nodes.

But if you are okay with dynamically generated usernames (no interactive user input) you can configure that in your workflow template.

tower-cli job launch --job-template=<template_id> --extra-vars="unique_name=my_unique_name_$(date +'%Y%m%d%H%M%S')"

This dynamically assigns a unique variable based on date and you can later retrieve them using jobid or using the unique name.

Hope it helps.

**Comment by Iyappanj** [comment]
Hi @Bobby-Math, Thanks for the suggestion. 

**Workflow templates do not support dynamically assigning unique names** - this is fine but is there a possibility of assigning a **label** for the Workflow Template dynamically during the execution ?

Also the label created with this module - https://docs.ansible.com/ansible/devel/collections/awx/awx/label_module.html#ansible-collections-awx-awx-label-module apply to all other nodes of the Workflow by default ?

**Comment by Bobby-Math** [comment]
@

> Hi @Bobby-Math, Thanks for the suggestion.
> 
> **Workflow templates do not support dynamically assigning unique names** - this is fine but is there a possibility of assigning a **label** for the Workflow Template dynamically during the execution ?
> 
> Also the label created with this module - https://docs.ansible.com/ansible/devel/collections/awx/awx/label_module.html#ansible-collections-awx-awx-label-module apply to all other nodes of the Workflow by default ?

The workflow template once configured is the set and cannot be modified. It is a complex job orchestration tool that could dynamically assign unique labels to *jobs* on it's own without relying on manual input. I am not sure about the other features. I think the label you apply to the nodes depending on how you configure it in the template. Only the nodes configured in the module should be affected.

Feel free to ask more or the details about the particular use cases of assigning labels to work templates etc..

**Comment by Iyappanj** [comment]
Hi @Bobby-Math Thanks for taking time in guiding me through it.

1. assigning labels to nodes is fine, But I want to assign label at the workflow template level dynamically at the run time using extra_vars. Because later I can just search for this label and should be able to get the workflow which was executed with the value. So that I can get details from all nodes using the workflow template itself.
2. when we check this [job template](https://docs.ansible.com/ansible/devel/collections/awx/awx/job_template_module.html#ansible-collections-awx-awx-job-template-module) it has "ask_labels_on_launch" parameter, which can be passed as extra_vars. but when going for a workflow template I am not assigning a playbook to it rather I just setup the connectivity between nodes and run them in order. So not sure how to get the dynamic labels attached to it.
3. I also came across this [label module](https://docs.ansible.com/ansible/devel/collections/awx/awx/label_module.html#ansible-collections-awx-awx-label-module) but not sure how this can help me assign the labels to whole workflow ?

**Comment by jpic** [comment]
I suppose most users don't use the labels, I'm sitting in front of an AWX with thousands of job templates like `[PRD][ORACLE][DEV] Post-refresh` or something.

Instead I created a single template with a single playbook and a role variable, and specifying the role of the job with labels, but users still fail to see them because they don't use expanded view and are not used to labels.

As such, I believe labels should be visible in default view when specified in the job because they give the information that we would unless have to specify by copying job templates.

**Comment by dberardo-com** [comment]
i dont know if this has been discussed, but in my case i have one template that has one variable that changes depending on the customer to which a specific piece of hardware is assigned. 

i have instructed the support team to go into the jobs page and search for a JOB with a given ID that they know includes the variables assigned to that specific customer. So they find the job an relaunch it, without having to go through the template creation process which my be error-prone.

in this scenario it would be very helpful if i could start the jobs from the template by giving them a name suffix which is the customer-s name so that the support people dont have to remember IDs but human friendly name when searching for jobs ... does it make sense ?

is this already achievable ?
